src/container/srv_target.c

```diff
@@ -524,6 +549,51 @@ ds_cont_local_open(uuid_t pool_uuid, uuid_t cont_hdl_uuid, uuid_t cont_uuid,
 		*cont_hdl = hdl;
 	}
 
+	/* It is possible to sync DTX status before destroy the CoS for close
+	 * the container. But that may be not enough. Because the server may
+	 * crashed before closing the container. Then the DTXs' status in the
+	 * CoS cache will be lost. So we need to re-sync the DTXs status when
+	 * open the container for the first time (not for cached open handle).
+	 *
+	 * On the other hand, even if we skip the DTX sync before destroy the
+	 * CoS cache when close the container, resync DTX when open container
+	 * is enough to guarantee related data records' visibility. That also
+	 * simplify the DTX logic.
+	 *
+	 * XXX: The logic is related with DAOS server re-intergration, but we
+	 *	do not support that currently. Then resync DTX when container
+	 *	open will be used as temporary solution for DTX related logic.
+	 *
+	 * We do not trigger dtx_resync() when start the server. Because:
+	 * 1. Currently, we do not support server re-integrate after restart.
+	 * 2. A server may has multiple pools and each pool may has multiple
+	 *    containers. These containers may not related with one another.
+	 *    Make all the DTXs resync together during the server start will
+	 *    cause the DTX resync time to be much longer than resync against
+	 *    single container just when use (open) it. On the other hand, if
+	 *    some servers are ready for dtx_resync, but others may not start
+	 *    yet, then the ready ones may have to wait or failed dtx_resync.
+	 *    Both cases are not expected.
+	 */
+	if (cont_uuid != NULL) {
+		struct ds_dtx_resync_args	*ddra = NULL;
+
+		D_ALLOC_PTR(ddra);
+		if (ddra == NULL)
+			D_GOTO(err_cont, rc = -DER_NOMEM);
+
+		ddra->po_hdl = hdl->sch_pool->spc_hdl;
```

Comment:
we are not taking refcount on ds_pool_child, and this is a background ULT, which means we might use a invalid pool handle if user close the container and disconnect the pool?

Reply:
OK,  I will fix it.